### ISSUE TITLE
bunny is greenish

### ISSUE THREAD

**Issue by FannieZhao0317** [alecjacobson/computer-graphics-ray-casting#24]
![image](https://user-images.githubusercontent.com/55219975/65562636-14ee9c80-df15-11e9-9b4f-7da2b038267b.png)
sphere and triangle look just fine..

**Comment by abhimadan** [comment]
This is likely related to how you're calculating your triangle normals. The triangles are given in a specific order to ensure that their normals are consistently ordered; computing cross products in a different way than what the given order suggests can make them point the opposite way (and thus have a different colour).

In a somewhat roundabout way, you can also notice this by looking at the plane colours. You can immediately tell which direction their normals are pointing, and comparing their colours with the colours on the triangles can give you a sense of which direction the triangle normals are pointing. Given that, what can you tell about the triangle normals in your image versus the triangle normals in the expected image?

**Comment by FannieZhao0317** [comment]
> This is likely related to how you're calculating your triangle normals. The triangles are given in a specific order to ensure that their normals are consistently ordered; computing cross products in a different way than what the given order suggests can make them point the opposite way (and thus have a different colour).
> 
> In a somewhat roundabout way, you can also notice this by looking at the plane colours. You can immediately tell which direction their normals are pointing, and comparing their colours with the colours on the triangles can give you a sense of which direction the triangle normals are pointing. Given that, what can you tell about the triangle normals in your image versus the triangle normals in the expected image?

Right! I noticed that the normal image for triangle.cpp is also yellowish. Thank you so much for reminding me about the direction of normal vectors!
